### PR TITLE
[ONNX] Update onnx submodule

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Update onnx submodule to 1.8 release

### CODE REVIEW

**Comment by spandantiwari** [comment]
Hi @houseroad - can you please take a look at this PR. It is updating some onnx+caffe2 tests which is needed for updadting the ONNX submodule to the latest ONNX 1.8 release. Thanks!
cc: @bzinodev

**Comment by neginraoof** [comment]
@shubhambhokare1
There's one more test failure related to Caffe2: test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_small_model

Also, could you please check the pytorch_bazel_build failure and if it's related?

Thanks!

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def forward(self, x):
                 return x
 
         x = np.random.rand(2, 3, 10, 10).astype("float32")
-        self.generic_test(SimpleModel(), (x,), input_names=["x"], relaxed_check=True)
+        # self.generic_test(SimpleModel(), (x,), input_names=["x"], relaxed_check=True)
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2_quantized.py:283]: wait we cannot completely disable this...

**Review by houseroad — COMMENTED** [review]
Could we fix the bazel build?

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_caffe2_quantized.py:283]: This is failing the onnx shape inference call, which was suppressing all errors in previous onnx releases. The error was not from within pytorch. 

Is it ok in this PR to surround the shape inference call with try catch? that would preserve the same status.

**Comment by neginraoof** [comment]
@houseroad I think we need help fixing the bazel failure. The error is:
```external/onnx/onnx/onnx-data_pb.h:7:10: fatal error: onnx/onnx-data.pb.h: No such file or directory```

Can you help us understand if we need to update onnx.BUILD file to fix this?

**Comment by neginraoof** [comment]
@houseroad It would be great if you can give us contacts on Caffe2 side.

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 2]
```diff
@@ -416,9 +416,14 @@ void ONNXShapeTypeInference(Node* n, int opset_version) {
...
       UpdateOutputTypeByONNXProto(n, clone_node, *model_proto, symbol_map);
-    } catch (onnx::InferenceError& ex) {
+    } catch (std::runtime_error& ex) {
       // TODO: include this as warning once we have a more consolidated warning
       // system.
+      const char shape_err[] = "ShapeInferenceError";
+      const char type_err[] = "TypeInferenceError";
+      if ((strstr(ex.what(), shape_err) == NULL) &&
+          (strstr(ex.what(), type_err) == NULL))
+        throw std::runtime_error(ex.what());
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:426]: nit: try with just `throw;`. This should rethrow preserving whatever exception it got.

File: test/onnx/test_pytorch_onnx_caffe2_quantized.py [hunk 3]
```diff
@@ -280,7 +280,11 @@ def forward(self, x):
                 return x
 
         x = np.random.rand(2, 3, 10, 10).astype("float32")
-        self.generic_test(SimpleModel(), (x,), input_names=["x"], relaxed_check=True)
+        try:
+            self.generic_test(SimpleModel(), (x,), input_names=["x"], relaxed_check=True)
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_caffe2_quantized.py:284]: it'd be better if the try catch is precisely around the shape_inference call ..

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks!

**Review by BowenBao — APPROVED** [review]
LGTM, thanks. @houseroad all the CI issues have been addressed.

**Review by bzinodev — APPROVED** [review]

**Review by houseroad — APPROVED** [review]
LGTM. Thanks for keeping c2 tests :-)

**Comment by spandantiwari** [comment]
Thank you @houseroad and @jspisak for your help!

**Pull request closed without merging**